server/lib/scene/scene.duplicate.js

```diff
@@ -0,0 +1,46 @@
+const db = require('../../models');
+const { NotFoundError } = require('../../utils/coreErrors');
+
+/**
+ * @description Duplicate a scene.
+ * @param {string} selector - The selector of the source scene.
+ * @param {string} name - The name of the duplicated scene.
+ * @param {string} icon - The icon of the duplicated scene.
+ * @returns {Promise} - Resolve with the scene.
+ * @example
+ * duplicateScene({
+ *  selector: 'test'
+ * });
+ */
+async function duplicate(selector, name, icon) {
+  const existingScene = await db.Scene.findOne({
+    where: {
+      selector,
+    },
+  });
+
+  if (existingScene === null) {
+    throw new NotFoundError('Scene not found');
+  }
+
+  const plainExistingScene = existingScene.get({ plain: true });
+
+  const newScene = {
+    name,
+    icon,
+    active: plainExistingScene.active,
+    actions: plainExistingScene.actions,
+    triggers: plainExistingScene.triggers,
+  };
+
+  const createdScene = await db.Scene.create(newScene);
+  const plainScene = createdScene.get({ plain: true });
+  // add scene to live store
+  this.addScene(plainScene);
+  // return created scene
+  return plainScene;
```

Comment:
Just a small feedback, maybe it could have been wise here simply to call: 

```
return this.create(newScene);
```

? 

This way we avoid duplicating code, and if we add new things to the scene.create function, we avoid having to copy-paste code here :)

Reply:
Yes, you are completely right 👍